### DISCUSSION TITLE
Socks5h

### DISCUSSION THREAD

**Discussion by sam-kleiner** [encode/httpcore#673] [Ideas]
I would like to add support for socks5h. socks5h:// means that the hostname is resolved by the SOCKS server. socks5:// means that the hostname is resolved locally. [See similar issue in urllib3](https://github.com/urllib3/urllib3/issues/1035)

I already made a PR here https://github.com/encode/httpcore/pull/672, because the link to discussion groups was broken.

**Comment by sam-kleiner** [comment]
Oh actually this might be an issue. When looking at httpx discussion board I found this. https://github.com/encode/httpx/discussions/2592. Which after some testing it looks like I can confirm socks5:// is doing resolution on this SOCKS server, which to my knowledge should only happen when using socks5h://. Is there a different option anywhere to enable or disable host resolution?

**Comment by lovelydinosaur** [comment]
> Is there a different option anywhere to enable or disable host resolution?

Currently not, no.
